sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -361,13 +375,19 @@ def set_xml_body(self, data):
     def set_json_body(self, data):
         """Set a JSON-friendly object as the body of the request.
 
+        Sets your content type header to "application/json" if you
+        didn't input any content type headers
+
         :param data: A JSON serializable object
+        :raises: TypeError
         """
         if data is None:
             self.data = None
         else:
             self.data = json.dumps(data)
             self.headers["Content-Length"] = str(len(self.data))
+        if not self.headers.get("content-type"):
+            self.headers["Content-Type"] = "application/json"
```

Comment:
"content-type"? "Content-Type"?

Reply:
headers is a case insensitive dict, so searching with case doesn't matter. but you're right, it does look a little weird to just randomly switch casing from line to line, so will make 389 capitalized as well, thanks!